Client/TabManagement/TabManagerImplementation.swift

```diff
@@ -301,10 +301,13 @@ class TabManagerImplementation: LegacyTabManager, Notifiable {
     // MARK: - Select Tab
 
     override func selectTab(_ tab: Tab?, previous: Tab? = nil) {
+        let url = tab?.url
         guard shouldUseNewTabStore(),
               let tab = tab,
               let tabUUID = UUID(uuidString: tab.tabUUID)
         else {
+            willSelectTab(url)
+            defer { didSelectTab(url) }
```

Comment:
The defer here doesn't really offer any convenience and just makes it a little harder to reason about, I'd suggest just moving the didSelectTab call to directly before the return.

Reply:
Fixed ✅